packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx

```diff
@@ -1,58 +1,123 @@
 import { ChevronDown } from "@comet/admin-icons";
-import { Button, ComponentsOverrides, Theme } from "@mui/material";
-import { ButtonProps } from "@mui/material/Button";
-import { WithStyles } from "@mui/styles";
-import withStyles from "@mui/styles/withStyles";
-import clsx from "clsx";
+import { Button, buttonClasses, ButtonProps, ComponentsOverrides, svgIconClasses } from "@mui/material";
+import { css, styled, Theme } from "@mui/material/styles";
+import { useThemeProps } from "@mui/system";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 
 import { FilterBarActiveFilterBadge, FilterBarActiveFilterBadgeProps } from "../filterBarActiveFilterBadge/FilterBarActiveFilterBadge";
-import { FilterBarButtonClassKey, styles } from "./FilterBarButton.styles";
 
 /**
  * @deprecated Use MUI X Data Grid in combination with `useDataGridRemote` instead.
  */
-export interface FilterBarButtonProps extends ButtonProps {
+
+export type FilterBarButtonClassKey = "root" | "open" | "hasDirtyFields" | "filterBadge";
+
+type OwnerState = Pick<FilterBarButtonProps, "openPopover" | "numberDirtyFields">;
+
+const Root = styled(Button, {
+    name: "CometAdminFilterBarButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.root, ownerState.openPopover && styles.open, ownerState.numberDirtyFields && styles.numberDirtyFields];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ theme, ownerState }) => css`
+        position: relative;
+        cursor: pointer;
+        display: flex;
+        background-color: ${theme.palette.common.white};
+        border-color: ${theme.palette.grey[100]};
+        border-radius: 2px;
+
+        & .${buttonClasses.startIcon} .${svgIconClasses.root}, & .${buttonClasses.endIcon} .${svgIconClasses.root} {
```

Comment:
I don't think the `&` is necessary here... or is it required for higher specificity? 🤔 

```suggestion
        .${buttonClasses.startIcon} .${svgIconClasses.root}, .${buttonClasses.endIcon} .${svgIconClasses.root} {
```

Reply:
I corrected it to && as it is needed to overwrite the styling applied in MuiButton.ts